### PR TITLE
Clarify purpose of rerun-if-changed in build.rs

### PR DESCRIPTION
**Pull request by adamgreig**
The current `build.rs` contains
```rust
    println!("cargo:rerun-if-changed=build.rs");
    println!("cargo:rerun-if-changed=memory.x");
```

This causes the build script to be re-run if (and *only if*) `build.rs` or `memory.x` change. The line for `build.rs` is redundant: the Cargo guide [says](https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script) the build script is always rerun when `build.rs` changes, so we can drop that line.

The remaining line caused me a bit of confusion when adding other functionality to my project's build script, before I realised it tells Cargo to *only* rerun the build script when `memory.x` changes, as opposed to the default behaviour which is to rerun it on every build. The comment helps clarify the point of this line, so if users add other functionality to their build scripts it is hopefully easier to notice.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -12,6 +12,6 @@ fn main() {
         .unwrap();
     println!("cargo:rustc-link-search={}", out.display());
 
-    println!("cargo:rerun-if-changed=build.rs");
+    // Only re-run the build script when memory.x is changed, instead of at every build
```
Comment (japaric) [inline build.rs]: Nice to know that `rerun-if-changed=build.rs` is no longer needed!

"at every build" is not correct.

If `rerun-if-changed` is not present the build script only runs when any file within the Cargo project source (i.e. not in the `target` directory) changes. `cargo build; cargo build` will only run the build script *once*, **unless** the build script is modifying something in the Cargo project source, which is something you should avoid doing exactly because it causes the build script to fire on every `cargo build`. Like the guide says build scripts should put any generated files / output in `OUT_DIR` (which resides within the `target` directory).

TL;DR this should say something like "instead of when any part of the source code changes"

Comment (adamgreig) [reply build.rs]: You're right, I meant "whenever any of the source code changes" instead of "at every build". For context in this case the build script was using [built](https://crates.io/crates/built) to generate build-time information including the git commit and dirty flag; obviously this needs the build script to actually run every time the source code changes.
